src/fprime/cookiecutter_templates/cookiecutter-fprime-component/{{cookiecutter.component_name}}/{{cookiecutter.component_name}}ComponentAi.xml

```diff
@@ -0,0 +1,85 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<component name="{{cookiecutter.component_name}}" kind="{{cookiecutter.component_kind}}" namespace="TEMP_NAMESPACE">
+    <!-- Example of importing a port type -->
+{% if cookiecutter.ports == "yes" %}
+    <import_port_type>Svc/Ping/PingPortAi.xml</import_port_type>
+{% endif %}
+    <comment>{{cookiecutter.component_short_description}}</comment>
+{% if cookiecutter.ports == "yes" %}
+    <ports>
+        <!-- Input ports must be of kind "sync_input" or "async_input" or "guarded_input" -->
+        <!-- Input ping port -->
+{%- if cookiecutter.component_kind == 'passive' %}
+        <port name="PingIn" data_type="Svc::Ping" kind="sync_input" max_number="1">
+{%- else %}
+        <port name="PingIn" data_type="Svc::Ping" kind="async_input" max_number="1">
+{%- endif %}
+            <comment>
+            The ping input port
+            </comment>
+        </port>
+        <!-- One kind of output port -->
+        <!-- Output ping port -->
+        <port name="PingOut" data_type="Svc::Ping" kind="output" max_number="1">
+            <comment>
+            The ping output port
+            </comment>
+        </port>
+    </ports>
+{% endif %}
+{% if cookiecutter.commands == "yes" %}
+    <commands>
+{%- if cookiecutter.component_kind == 'passive' %}
+        <command kind="sync" opcode="0" mnemonic="PR_StopPings">
+{%- else %}
+        <command kind="async" opcode="0" mnemonic="PR_StopPings">
+{%- endif %}
+            <comment>
+            Command to disable ping response
+            </comment>
+        </command>
+    </commands>
+{% endif %}
+{% if cookiecutter.parameters == "yes" %}
+    <parameters>
+        <parameter id="5" set_opcode="6" save_opcode="1" name="parameter1" data_type="U32" default="10">
+            <comment>
+            A test parameter
+            </comment>
+        </parameter>
+        <parameter id="7" set_opcode="8" save_opcode="3" name="parameter2" data_type="I16" default="11">
+            <comment>
+            A test parameter
+            </comment>
+        </parameter>
+    </parameters>
+{% endif %}
+{% if cookiecutter.telemetry == "yes" %}
+    <telemetry>
+        <channel id="0" name="PR_NumPings" data_type="U32">
+            <comment>
+            Number of pings received
+            </comment>
+        </channel>
+    </telemetry>
+{% endif %}
+{% if cookiecutter.events == "yes" %}
+    <events>
+        <event id="0" name="PR_PingsDisabled" severity="ACTIVITY_HI" format_string = "PingReceiver ping responses disabled" >
```

Comment:
I think that's a good idea, but I also like keeping two examples here because I feel like it shows the optional argument well. Is it okay to just have two generic examples here?

Reply:
It does do a good job of demonstrating the optional argument here, but is this the place to demonstrate that? I feel something like that better belongs in the documentation. I think demonstrating this every time somebody generates a new component is a little excessive, especially if it requires users to remove autogenerated code they're not using.

Reply:
That is a good point. Especially because the cookiecutter won't only be used for learning purposes. I'll remove one of them then